### PR TITLE
Add details popup window for the Portfolio project

### PR DESCRIPTION
**Pull request by lenkon**
Project name: Portfolio - details popup window

Project description: For this milestone of the portfolio website, a popup window will be implemented that includes the project details in both desktop and mobile. In order to do this, information about the projects will be stored in a JavaScript object.

Project requirements: 
Implement the following features-
- When the user clicks (or taps) the button to check project details, the popup with details about the project appears.
- When the user clicks (or taps) the close (X) button, the popup disappears.
- In order to associate each project with the popup window details, refactor the project section:
- Use a JavaScript array to store all of the information for all projects.
- For each project store the following pieces of data in a JavaScript object, at a minimum: name, description, featured image, technologies, link to live version, link to source.
- Update the main page so that the projects section is created dynamically using the information stored in that JavaScript object. Remember, all of the HTML in that section is created when the page loads.
- After that, implement the popup window.
You should implement popups for both mobile and desktop screen sizes

Project requirements link: https://github.com/microverseinc/curriculum-html-css/blob/main/portfolio/8_popup.md

### CODE REVIEW

File: js/script.js [hunk 1]
```diff
@@ -25,4 +25,206 @@ menuClose.addEventListener('click', closeMenu);
...
+        <nav class="popup-card-details">
+          <h2 class="card-title-text">${item.title}</h2>
+          <button id="popup-close-button" class="popup-close-button">
+            <img src="./images/Popup-close-Icon.png" alt="popup close icon">
+          </button>
+        </nav>
+        <div class="card-list">
+          ${roles.join('')}
+        </div>
+      </header>
```
Comment (thecodechaser) [inline js/script.js:160]: - [ ] You did a great job on the styles but for this project, you need to stick to the [Figma](https://www.figma.com/file/l7SqJ3ZfkAKih9sFxvWSR4/Microverse-Student-Project-1?node-id=0%3A1&t=Z2va5AFebQpvwsk5-0) design as much as possible. For desktop screens, Please make sure to increase the font size for the project title, and tags and use the blur effect in the model background.

### Your design

image

### Figma design

image

File: js/script.js [hunk 2]
```diff
@@ -25,4 +25,206 @@ menuClose.addEventListener('click', closeMenu);
...
+      <header class="popup-details-header">
+        <nav class="popup-card-details">
+          <h2 class="card-title-text">${item.title}</h2>
+          <button id="popup-close-button" class="popup-close-button">
+            <img src="./images/Popup-close-Icon.png" alt="popup close icon">
+          </button>
+        </nav>
+        <div class="card-list">
+          ${roles.join('')}
+        </div>
```
Comment (thecodechaser) [inline js/script.js:159]: - [ ] For desktop screens, Please make sure to reduce the button height and increase the width to match the Figma design.

### Your design

image

### Figma design

image

Comment (thecodechaser) [inline js/script.js:159]: - [ ] For mobile screens, Please make sure to add some margin to the left and bottom of the model and the model buttons should match the Figma design.

### Your design

image

### Figma design

image

**Review by thecodechaser — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️ 

## Hi @lenkon, and @Yemisirach👋, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## To Highlight 🎉
- No linter errors. :heavy_check_mark:
- GitHub flow is followed. :heavy_check_mark:
- Professional README file. :heavy_check_mark:
- PR has a good title and summary. ✔️



## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: js/script.js (hunk 1 above)
Comment (lenkon) [reply js/script.js:160]: Hi @thecodechaser, thanks for the valuable review. I have made the required changes in the modal background and font sizes. Please go through it and let me know if you need any changes.

File: js/script.js (hunk 2 above)
Comment (lenkon) [reply js/script.js:159]: Hi @thecodechaser, thanks for the feedback. I have made the necessary changes to the button dimensions. Please go through it and let me know if you need anything else.

Comment (lenkon) [reply js/script.js:159]: Hi @thecodechaser, thanks for the valuable feedback. I have added the required margins to the modal and also made changes to the buttons.

**Review by TuyishimireEric — APPROVED** [review]
a# Status: APPROVED 🟢  
### Hi  @lenkon  👋 👋 

#### I'm @TuyishimireEric, Your code reviewer,
Good job so far! 🥇 
done

Your project is complete!  There is nothing else to say other than... it's time to merge it 🦾  
Congratulations! 🎉 

## Highlights
✔️Professional Readme 👍 
✔️Professional PR 👍 
✔️No Linter error👍 

### Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**